### DISCUSSION TITLE
Updated certificates, new log, and STV

### DISCUSSION THREAD

**Discussion by skovatch** [Ascoware/get-iplayer-automator#438] [Announcements]
- Fixed bug where empty rows in the download queue stopped all downloading
- Updated root certificates, which may or may not reduce connection errors when fetching program listings
- Introduced new logging code. The log window will now highlight warnings and errors in color.
- Individual, non-DRM programs can now be downloaded from stv.tv using 'Use Current Webpage'. 

<hr /><em>This discussion was created from the release <a href='https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.24.5'>Updated certificates, new log, and STV </a>.</em>

**Comment by skovatch** [comment]
Until I can find a reliable means of scraping STV content you won't be able to search for or auto-record STV shows. Assuming I can do that I may drop ITV support altogether, even though not everything on ITV makes its way to STV.

**Comment by Temblorz** [comment]
Just thought I'd let you know that something in this version makes it impossible to use with OS X 10.13 (High Sierra), in case that matters to anyone but me.  :-)

**Comment by SOLOSAIR** [comment]
What happened to the blue refresh button?

stv 'use current webpage' results in status of 'not in cache' even after using 'file/check for cache update' (screenshot)

edit: discovered 'start' enabled successful download without being added to apple tv app.![image](https://user-images.githubusercontent.com/37366634/212781459-e9b6f1ae-fcc8-4f14-be67-5fa8c634c206.jpeg)

**Reply by skovatch** [reply]
STV programs are not cached (see the first comment above), so Process PID won't have any effect.

**Reply by SOLOSAIR** [reply]
thanks.  How do I get the blue refresh button back please?

**Reply by skovatch** [reply]
I just pushed an update to fix the tagging issue, but any STV downloads can be found in the "Home Movies" section of TV.app until you update.

I don't understand what you mean by the 'blue refresh button'. I don't recall one ever existing.

**Comment by SOLOSAIR** [comment]
thanks. I regularly used a round blue button along with green 'start', red 'stop', 'series-link to queue, 'pvr' 'show log' and 'preferences' round buttons at the top of the app, to manually refresh the cache - the equivalent to 'check for cache update' under the file menu.    I used it often as sometimes GiA was kept open for long periods of time while I used other apps and when I came back to GiA I hit the blue button instead of closing and opening the app to auto refresh the cache.
  
  

  
>

**Reply by skovatch** [reply]
Oh, I see it now... it's an additional button hiding in the toolbar. At some point something disabled the ability to customize the toolbar -- I don't remember explicitly turning that off. I can add it back for the next release. In the meantime, cmd-R does the same thing as the button.

**Reply by SOLOSAIR** [reply]
many, many thanks for reintroducing blue refresh button availability.  very much appreciated, as is ALL the work you do on GiA 🙂

**Comment by SOLOSAIR** [comment]
handy to know that, thanks very much
  
  
  
  
  
  
  

  
  
>
